database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -284,29 +277,30 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
                                 buildUpdatedSummary:
                                 (Option[WorkflowMetadataSummaryEntry], Seq[MetadataEntry])
                                   => WorkflowMetadataSummaryEntry,
-                                summaryPositionFunction: Seq[MetadataEntry] => DBIO[Long],
+                                summaryPositionFunction: Seq[MetadataEntry] => Long,
                                 summaryName: String
                                )(implicit ec: ExecutionContext): DBIO[Long] = {
+
+    val exactMatchMetadataKeys = List(
+      startMetadataKey, endMetadataKey, nameMetadataKey, statusMetadataKey, submissionMetadataKey, parentWorkflowIdKey, rootWorkflowIdKey)
+    val startsWithMetadataKeys = List(labelMetadataKey)
+
     for {
-      metadataEntries <- dataAccess.metadataEntriesForIdRange((
+      rawMetadataEntries <- dataAccess.metadataEntriesForIdRange((
         minMetadataEntryId,
-        maxMetadataEntryId,
-        startMetadataKey,
-        endMetadataKey,
-        nameMetadataKey,
-        statusMetadataKey,
-        submissionMetadataKey,
-        parentWorkflowIdKey,
-        rootWorkflowIdKey,
-        labelMetadataKey
+        maxMetadataEntryId
       )).result
+      summaryPosition = summaryPositionFunction(rawMetadataEntries)
+      metadataEntries = rawMetadataEntries filter { entry =>
+        entry.callFullyQualifiedName.isEmpty && entry.jobIndex.isEmpty && entry.jobAttempt.isEmpty &&
+          (exactMatchMetadataKeys.contains(entry.metadataKey) || startsWithMetadataKeys.exists(entry.metadataKey.startsWith))
+      }
       metadataWithoutLabels = metadataEntries
-        .filterNot(_.metadataKey.contains(labelMetadataKey))
+        .filterNot(_.metadataKey.contains(labelMetadataKey)) // Why are these "contains" while the filtering is "starts with"?
```

Comment:
I think this and the similar construct two lines down are unrelated bugs but I could be convinced otherwise. 😛

Reply:
I'm inclined to agree that this should also be `startsWith`.

I think we're probably only doing ok today because there happen to be no metadata keys with `labels:` in them